src/hesiod.c

```diff
@@ -0,0 +1,152 @@
+  return hsk_hesiod_txt_push(name, number, an);
+}
+
+static bool
+hsk_hesiod_txt_push_hash(char *name, uint8_t *data, hsk_dns_rrs_t *an) {
+  char hash[65];
+  for (int i = 0; i < 32; i++)
+    sprintf(&hash[i * 2], "%02x", data[i]);
+  return hsk_hesiod_txt_push(name, hash, an);
+}
+
+static bool
+hsk_hesiod_txt_push_float(char *name, float f, hsk_dns_rrs_t *an) {
+  char value[65];
+  sprintf(value, "%f", f);
+  return hsk_hesiod_txt_push(name, value, an);
+}
+
+static bool
+hsk_hesiod_txt_push_peer(hsk_peer_t *peer, hsk_dns_rrs_t *an) {
+  char label[65];
+  char sublabel[40];
+  sprintf(sublabel, "%llu.peers.pool.hnsd.", peer->id);
+
+  sprintf(label, "host.%s", sublabel);
+  if (!hsk_hesiod_txt_push(label, peer->host, an))
+    return false;
+
+  sprintf(label, "agent.%s", sublabel);
+  if (!hsk_hesiod_txt_push(label, peer->agent, an))
+    return false;
+
+  return true;
+}
+
+hsk_dns_msg_t *
+hsk_hesiod_resolve(hsk_dns_req_t *req, hsk_ns_t *ns) {
+  hsk_dns_msg_t *msg = hsk_dns_msg_alloc();
+
+  if (!msg)
+    return NULL;
+
+  msg->flags |= HSK_DNS_AA;
+  hsk_dns_rrs_t *an = &msg->an;
+
+  // CHAIN
+  //   TIP
+  if (hsk_dns_is_subdomain(req->name, "hash.tip.chain.hnsd.")) {
+    if (!hsk_hesiod_txt_push_hash("hash.tip.chain.hnsd.",
+                                  ns->pool->chain.tip->hash,
+                                  an))
+      goto fail;
+  }
+
+  if (hsk_dns_is_subdomain(req->name, "height.tip.chain.hnsd.")) {
+    if (!hsk_hesiod_txt_push_u64("height.tip.chain.hnsd.",
+                                 ns->pool->chain.tip->height,
+                                 an))
+      goto fail;
+  }
+
+  if (hsk_dns_is_subdomain(req->name, "time.tip.chain.hnsd.")) {
+    if (!hsk_hesiod_txt_push_u64("time.tip.chain.hnsd.",
+                                 ns->pool->chain.tip->time,
+                                 an))
+      goto fail;
+  }
+
+  if (hsk_dns_is_subdomain(req->name, "synced.chain.hnsd.")) {
+    if (!hsk_hesiod_txt_push("synced.chain.hnsd.",
+                             ns->pool->chain.synced ? "true" : "false",
+                             an))
+      goto fail;
+  }
+
+  //   PROGRESS
+  if (hsk_dns_is_subdomain(req->name, "progress.chain.hnsd.")) {
+    if (!hsk_hesiod_txt_push_float("progress.chain.hnsd.",
+                             hsk_chain_progress(&ns->pool->chain),
+                             an))
+      goto fail;
+  }
+
+  // POOL
+  //  SIZE
+  if (hsk_dns_is_subdomain(req->name, "size.pool.hnsd.")) {
+    char size[65];
+    sprintf(size, "%d", ns->pool->size);
+    if (!hsk_hesiod_txt_push("size.pool.hnsd.",
+                             size,
+                             an))
+      goto fail;
+  }
+
+  //  PEERS
+  if (hsk_dns_is_subdomain(req->name, "peers.pool.hnsd.")) {
+    hsk_peer_t *peerIter, *next;
+    for (peerIter = ns->pool->head; peerIter; peerIter = next) {
+      if (!hsk_hesiod_txt_push_peer(peerIter, an))
+        goto fail;
+
+      next = peerIter->next;
+    }
+  }
```

Comment:
it would be great if instead of peer id, a normal counter is used. Peer ID itself isn't really useful since the ip:port can act as an identifier and a parser's complexity increases when the numbers are non-deterministic.

Technically the parser can get around this by _ignoring_ the number and just pushing onto an array, but imo it's cleaner if the numbers in the names are in order and consecutive.

Otherwise it ends up looking like:
```js
{
  chain: {
    tip: {
      hash: '0000000000000004289e32985c3325ccfcafe6facbbdf3d77908cd2787c6e55c',
      height: '145177',
      time: '1667733359'
    },
    synced: 'true',
    progress: '1.000000'
  },
  pool: { size: '2', peers: [ {}, {}, [Object], [Object] ] }
}
```

Reply:
added in 6e8847551d2978a946e173e27c733d652bbc69de